pjsip/include/pjsip-simple/dialog_info.h

```diff
@@ -0,0 +1,353 @@
+
+    const pj_str_t * (*get_call_id)(const pjsip_dlg_info_dialog *);
+    void (*set_call_id)(pj_pool_t *pool, const pjsip_dlg_info_dialog *,
+                        const pj_str_t *);
+
+    const pj_str_t * (*get_remote_tag)(const pjsip_dlg_info_dialog *);
+    void (*set_remote_tag)(pj_pool_t *pool, const pjsip_dlg_info_dialog *,
+                           const pj_str_t *);
+
+    const pj_str_t * (*get_local_tag)(const pjsip_dlg_info_dialog *);
+    void (*set_local_tag)(pj_pool_t *pool, const pjsip_dlg_info_dialog *,
+                          const pj_str_t *);
+
+    const pj_str_t * (*get_direction)(const pjsip_dlg_info_dialog *);
+    void (*set_direction)(pj_pool_t *pool, const pjsip_dlg_info_dialog *,
+                          const pj_str_t *);
+
+    const pj_str_t * (*get_state)(const pjsip_dlg_info_dialog *);
+    void (*set_state)(pj_pool_t *pool, const pjsip_dlg_info_dialog *,
+                      const pj_str_t *);
+
+    const pj_str_t * (*get_duration)(const pjsip_dlg_info_dialog *);
+    void (*set_duration)(pj_pool_t *pool, const pjsip_dlg_info_dialog *,
+                         const pj_str_t *);
+
+    pjsip_dlg_info_local* (*get_local)(const pjsip_dlg_info_dialog *);
+    pjsip_dlg_info_local* (*set_local)(pj_pool_t *pool,
+                                       const pjsip_dlg_info_dialog *,
+                                       const pjsip_dlg_info_local *);
+
+    pjsip_dlg_info_remote* (*get_remote)(const pjsip_dlg_info_dialog *);
+    pjsip_dlg_info_remote* (*set_remote)(pj_pool_t *pool,
+                                        const pjsip_dlg_info_dialog *,
+                                        const pjsip_dlg_info_remote *);
+} pjsip_dlg_info_dialog_op;
+
+typedef struct pjsip_dlg_info_dialog_info_op
+{
+    void (*construct)(pj_pool_t *, pjsip_dlg_info_dialog_info *,
+                     const pj_str_t *, const pj_str_t *, const pj_str_t *);
+
+    const pj_str_t * (*get_state)(const pjsip_dlg_info_dialog_info *);
+    void (*set_state)(pj_pool_t *pool, const pjsip_dlg_info_dialog_info *,
+                      const pj_str_t *);
+
+    const pj_str_t * (*get_version)(const pjsip_dlg_info_dialog_info *);
+    void (*set_version)(pj_pool_t *pool, const pjsip_dlg_info_dialog_info *,
+                        const pj_str_t *);
+
+    const pj_str_t * (*get_entity)(const pjsip_dlg_info_dialog_info *);
+    void (*set_entity)(pj_pool_t *pool, const pjsip_dlg_info_dialog_info *,
+                       const pj_str_t *);
+
+    pjsip_dlg_info_dialog* (*get_dialog)(const pjsip_dlg_info_dialog_info *);
+    void (*set_dialog)(pj_pool_t *pool, const pjsip_dlg_info_dialog_info *,
+                       const pjsip_dlg_info_dialog * );
+} pjsip_dlg_info_dialog_info_op;
+
+
+extern struct pjsip_dlg_info_op_desc
+{
+    pjsip_dlg_info_dialog_info  dialog_info;
+    pjsip_dlg_info_dialog       dialog;
+    pjsip_dlg_info_local        local;
+    pjsip_dlg_info_remote       remote;
+} pjsip_dlg_info_op;
```

Comment:
These structs do not seem to be used, remove them? Or will there be possible alternative implementations for `dlg_info` manipulations?